### PR TITLE
Show preview position before change it

### PR DESCRIPTION
**Pull request by neysidev**
When user hover on alignments items, will see the preview of the box on this position.

### CODE REVIEW

**Comment by pooridev** [comment]
Dope feature!
Thanks for the contribution!

File: src/components/code-editor-wrapper/index.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import Moveable from 'react-moveable'
 import { OnResizeStart, OnResize } from 'react-moveable/declaration'
 
 import CodeEditor from './code-editor'
+import PreviewNextPosition from './preview-next-position'
```
Comment (pooridev) [inline src/components/code-editor-wrapper/index.tsx:8]: next-position-preview is the correct form

File: src/components/code-editor-wrapper/index.tsx [hunk 2]
```diff
@@ -17,6 +18,7 @@ const CodeEditorWrapper = () => {
     <>
       <Box id='code-editor' ref={moveableTargetRef}>
         <CodeEditor />
+        <PreviewNextPosition />
```
Comment (pooridev) [inline src/components/code-editor-wrapper/index.tsx:21]: Change it to `NextPositionPreview` please

File: src/components/code-editor-wrapper/preview-next-position/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    case 'center-y':
+      return { top: '50%', left: `${left}%`, transform: `translate(-${left}%, -50%)` }
+    case 'bottom':
+      return { bottom: 0, left: `${left}%`, transform: `translate(-${left}%, -0%)` }
+    default:
+      return {}
+  }
+}
+
+export default function PreviewNextPosition() {
```
Comment (pooridev) [inline src/components/code-editor-wrapper/preview-next-position/index.tsx:26]: Change it to ‍`NextPositionPreview` please

File: src/components/code-editor-wrapper/preview-next-position/index.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  const top = useRecoilValue(topPosition)
+  const left = useRecoilValue(leftPosition)
+  const hovered = useRecoilValue(hoveredPosition)
+
+  if (!hovered) return null
+
+  const width = document.querySelector('[aria-label="code-editor"]')?.clientWidth
+  const height = document.querySelector('[aria-label="code-editor"]')?.clientHeight
```
Comment (pooridev) [inline src/components/code-editor-wrapper/preview-next-position/index.tsx:34]: This is not a reactive solution and should be our LAST option!
Could u find a better solution for accessing this element?

File: src/components/code-editor-wrapper/index.tsx (hunk 1 above)
Comment (neysidev) [reply src/components/code-editor-wrapper/index.tsx:8]: oh sure, i will change it.

File: src/components/code-editor-wrapper/preview-next-position/index.tsx (hunk 4 above)
Comment (neysidev) [reply src/components/code-editor-wrapper/preview-next-position/index.tsx:34]: I will try to find a better solution.

Comment (pooridev) [reply src/components/code-editor-wrapper/preview-next-position/index.tsx:34]: My recommended solution is to create a Context Provider for storing the Editor's Ref.

**Pull request merged**